lib/pos_remote_node_indexer.go

```diff
@@ -0,0 +1,231 @@
+package lib
+
+import "github.com/deso-protocol/core/bls"
+
+// RNIndexId is a custom string type used for identifying different types of remote node indices.
+type RNIndexId string
+
+// Constants for different types of remote node indices.
+const (
+	RNIndexId_Validator              RNIndexId = "VALIDATOR"
+	RNIndexId_ValidatorAttempted     RNIndexId = "VALIDATOR_ATTEMPTED"
+	RNIndexId_NonValidator_Outbound  RNIndexId = "NONVALIDATOR_OUTBOUND"
+	RNIndexId_NonValidator_Inbound   RNIndexId = "NONVALIDATOR_INBOUND"
+	RNIndexId_NonValidator_Attempted RNIndexId = "NONVALIDATOR_ATTEMPTED"
+)
+
+// RemoteNodeIndexer is a structure that holds information about all remote nodes and their indices.
+type RemoteNodeIndexer struct {
+	// AllRemoteNodes is a map storing all remote nodes by their IDs.
+	AllRemoteNodes map[RemoteNodeId]*RemoteNode
```

Comment:
Any reason you use RemoteNodeId and not PeerID as the key for this map? If a RemoteNodeId is a tuple of (PeerID, AttemptID), and if PeerID is unique for each node, then PeerID alone is sufficient and simpler to use as a unique identifier for each node.